File: problemsets/Codeforces/Python/A727.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
a,b=map(int,input().split())
r=[b]
while b>a:
  d=b%10
  if d==1:b//=10
  elif d%2:break
  else:b//=2
  r+=[b]
r=r[::-1]
print([f'YES\n{len(r)}\n{" ".join(map(str,r))}','NO'][a!=b])

File: problemsets/Codeforces/C++/A727.cpp
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
#include<bits/stdc++.h>

using namespace std;

typedef long long ll;

int main() {
  ios::sync_with_stdio(0);
  int a, b; cin >> a >> b;
  unordered_set<ll> vis;
  vector<ll> ret;
  function<bool(ll)> findPath = [&](ll u) {
    if (u > b) return false;
    vis.insert(u);
    if (u == b) { ret.emplace_back(u); return true; }
    bool found = false;
    if (!vis.count(u*10+1) && (found=findPath(u*10+1))) ret.emplace_back(u);
    else if (!vis.count(u*2) && (found=findPath(u*2))) ret.emplace_back(u);
    return found;
  };
  findPath(a);
  if (ret.size() == 0) { cout << "NO"; return 0; }
  reverse(begin(ret), end(ret));
  cout << "YES" << endl;
  cout << ret.size() << endl;
  for (ll &u: ret) cout << u << " ";
}